{
    "CVE_data_meta": {
        "ASSIGNER": "security-advisories@github.com",
        "ID": "CVE-2021-29564",
        "STATE": "PUBLIC",
        "TITLE": "Null pointer dereference in `EditDistance`"
    },
    "affects": {
        "vendor": {
            "vendor_data": [
                {
                    "product": {
                        "product_data": [
                            {
                                "product_name": "tensorflow",
                                "version": {
                                    "version_data": [
                                        {
                                            "version_value": "< 2.1.4"
                                        },
                                        {
                                            "version_value": ">= 2.2.0, < 2.2.3"
                                        },
                                        {
                                            "version_value": ">= 2.3.0, < 2.3.3"
                                        },
                                        {
                                            "version_value": ">= 2.4.0, < 2.4.2"
                                        }
                                    ]
                                }
                            }
                        ]
                    },
                    "vendor_name": "tensorflow"
                }
            ]
        }
    },
    "data_format": "MITRE",
    "data_type": "CVE",
    "data_version": "4.0",
    "description": {
        "description_data": [
            {
                "lang": "eng",
                "value": "TensorFlow is an end-to-end open source platform for machine learning. An attacker can trigger a null pointer dereference in the implementation of `tf.raw_ops.EditDistance`. This is because the implementation(https://github.com/tensorflow/tensorflow/blob/79865b542f9ffdc9caeb255631f7c56f1d4b6517/tensorflow/core/kernels/edit_distance_op.cc#L103-L159) has incomplete validation of the input parameters. The fix will be included in TensorFlow 2.5.0. We will also cherrypick this commit on TensorFlow 2.4.2, TensorFlow 2.3.3, TensorFlow 2.2.3 and TensorFlow 2.1.4, as these are also affected and still in supported range."
            }
        ]
    },
    "impact": {
        "cvss": {
            "attackComplexity": "HIGH",
            "attackVector": "LOCAL",
            "availabilityImpact": "LOW",
            "baseScore": 2.5,
            "baseSeverity": "LOW",
            "confidentialityImpact": "NONE",
            "integrityImpact": "NONE",
            "privilegesRequired": "LOW",
            "scope": "UNCHANGED",
            "userInteraction": "NONE",
            "vectorString": "CVSS:3.1/AV:L/AC:H/PR:L/UI:N/S:U/C:N/I:N/A:L",
            "version": "3.1"
        }
    },
    "problemtype": {
        "problemtype_data": [
            {
                "description": [
                    {
                        "lang": "eng",
                        "value": "CWE-476: NULL Pointer Dereference"
                    }
                ]
            }
        ]
    },
    "references": {
        "reference_data": [
            {
                "name": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-75f6-78jr-4656",
                "refsource": "CONFIRM",
                "url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-75f6-78jr-4656"
            },
            {
                "name": "https://github.com/tensorflow/tensorflow/commit/f4c364a5d6880557f6f5b6eb5cee2c407f0186b3",
                "refsource": "MISC",
                "url": "https://github.com/tensorflow/tensorflow/commit/f4c364a5d6880557f6f5b6eb5cee2c407f0186b3"
            }
        ]
    },
    "source": {
        "advisory": "GHSA-75f6-78jr-4656",
        "discovery": "UNKNOWN"
    }
}